src/main/java/core/basesyntax/Car.java

```diff
@@ -1,13 +1,78 @@
 package core.basesyntax;
 
+import java.util.ArrayList;
 import java.util.List;
+import java.util.Objects;
 
-/**
- * Make this class immutable. See requirements in the README file
- */
-public class Car {
-    private int year;
-    private String color;
-    private List<Wheel> wheels;
-    private Engine engine;
+public final class Car {
+
+    private final int year;
+    private final String color;
+    private final List<Wheel> wheels;
+    private final Engine engine;
+
+    public Car(int year, String color, List<Wheel> wheels, Engine engine) {
+        this.year = year;
+        this.color = color;
+        this.engine = (engine == null ? null : engine.clone());
+        List<Wheel> wheelsCopy = new ArrayList<>();
+        for (Wheel wheel : wheels) {
+            wheelsCopy.add(wheel.clone());
+        }
+        this.wheels = wheelsCopy;
+    }
+
+    public Car changeEngine(Engine newEngine) {
+        return new Car(year, color, new ArrayList<>(wheels), newEngine.clone());
+    }
+
+    public Car changeColor(String newColor) {
+        return new Car(year, newColor, new ArrayList<>(wheels), engine);
```

Comment:
```suggestion
        return new Car(year, newColor, new ArrayList<>(wheels), engine.clone());
```
Because you will have 2 cars with one engine